Updated DVL models to include current profiling field (for plugin)

dtdavi1 opened the pull request:
Insignificant commit for the most part--I added a waterTrackBins field to the DVL models in support of the current profiling incorporation into the WHOI DSL plugin.  If you use the latest nps_dev branch from our copy of the WHOI DSL repos (ds_msgs and ds_sim) and run any of the DVL demos (e.g., roslaunch nps_uw_sensors whoi_teledyne_whn.launch) you'll get the current profile messages (rostopic echo dvl/dvl_current).  

You'll still get the messages without merging this pull request, but the plugin will default to 1 bin (I think I put 10 in all of the models).

dtdavi1 committed:
Updated DVL models to include current profiling field (for plugin)

bsb808 commented:
The code looks fine.  It would be good if we could develop an introductory demo for the WHOI folks to understand the functionality.  I started a wiki here: [DVL Water Tracking](https://github.com/Field-Robotics-Lab/dave/wiki/DVL-Water-Tracking).

Could you make sure that it make sense and add some verbosity before merging?

In particular, do the velocities in the verification test make sense?  I'm not sure the coordinate frame of the reported velocities.  It would be good to explicitly define the reference frame.  Does each beam have a tf frame to reference?

Also, a bit off topic, but where in the Dvl message is it reported if the DVL has bottom lock?  Good to note that in the wiki.

bsb808 reviewed (changes requested):
Code is fine - just need more verbose example and description of functionality in the wiki.  See comment in the Conversation.

dtdavi1 commented:
@bsb808 I'll update the wiki shortly--it looks good, but I'll add a launch file for water tracking/current profiling that doesn't require editing.  I'll take a look at the verification.  I haven't done a ton of verification testing (I wanted to push the commit before the feature freeze), but the implementation is pretty simple.  The plugin already computes a "beam velocity" for each beam and each beam has a unit vector for its orientation, so beam-relative current is just the velocity inverse times the unit vector.

Bottom lock is indicated in the Dvl message by the track mode field.  There is no beam-specific indication other than the range field (I think beams without bottom contact will indicate -1, but don't quote me on that).

dtdavi1 committed:
Added current profiling demo launch file (no motion to the DVL)

Also added a currentProfileCoordMode tag to all of the SDF and
URDF files to allow publishing of current profiling solutions in
beam-relative coordinates (default) and instrument-relative
coordinates.

dtdavi1 commented:
@bsb808  I did a more thorough validation, and everything looks correct as far as I can tell.  I also added a "instrument-specific reporting" capability to the velocity profile code (the only affect to this repository is a new "currentProfileCoordMode" tag in all of the models).  I also verified that it was working correctly.  I updated the demo/tutorial as well.
